src/main/java/org/example/DAOs/ClientDAO.java

```diff
@@ -0,0 +1,115 @@
+        try (PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Client client = Client.builder()
+                        .id(resultSet.getLong("id"))
+                        .name(resultSet.getString("name"))
+                        .email(resultSet.getString("email"))
+                        .createdAt(resultSet.getTimestamp("created_at").toLocalDateTime())
+                        .updatedAt(resultSet.getTimestamp("updated_at").toLocalDateTime())
+                        .build();
+                return Optional.of(client);
+            }
+        } catch (SQLException e) {
+            throw new DAOException("Failed to find client by ID: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Client> getAll() {
+        String sql = "SELECT * FROM clients";
+        List<Client> clients = new ArrayList<>();
+        try (PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Client client = Client.builder()
+                        .id(resultSet.getLong("id"))
+                        .name(resultSet.getString("name"))
+                        .email(resultSet.getString("email"))
+                        .createdAt(resultSet.getTimestamp("created_at").toLocalDateTime())
+                        .updatedAt(resultSet.getTimestamp("updated_at").toLocalDateTime())
+                        .build();
+                clients.add(client);
+            }
+        } catch (SQLException e) {
+            throw new DAOException("Failed to retrieve all clients", e);
+        }
+        return clients;
+    }
+
+    @Override
+    public void save(Client client) {
+        String sql = "INSERT INTO clients (name, created_at, updated_at, email) VALUES (?, ?, ?, ?)";
+        try (PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setString(1, client.getName());
+            statement.setTimestamp(2, Timestamp.valueOf(client.getCreatedAt() != null ? client.getCreatedAt() : LocalDateTime.now()));
+            statement.setTimestamp(3, Timestamp.valueOf(client.getUpdatedAt() != null ? client.getUpdatedAt() : LocalDateTime.now()));
+            statement.setString(4, client.getEmail());
+            int rowsAffected = statement.executeUpdate();
+            if (rowsAffected == 0) {
+                throw new DAOException("Failed to save client: No rows affected");
+            }
+        } catch (SQLException e) {
+            throw new DAOException("Failed to save client: " + client, e);
+        }
+    }
+
+    public void update(Client client, Client updatedClient) {
+        String sql = "UPDATE clients SET name = ?, email = ?, updated_at = ? WHERE id = ?";
```

Comment:
so do we need as a param for method the whole object of previous(non updated) version?

Reply:
actually no, it's enough to use only object's id of previous version